File: bidappdemoapp-java/app/src/main/java/com/bidapp/demo/BannerViewDelegate.java
package com.bidapp.demo;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import io.bidapp.sdk.AdInfo;
import io.bidapp.sdk.BIDBannerViewDelegate;
import io.bidapp.sdk.BannerView;


public class BannerViewDelegate implements BIDBannerViewDelegate {



    @Override
    public void adViewDidLoadAd(@NonNull BannerView bannerView, @Nullable AdInfo adInfo) {
        System.out.println("App - adViewDidLoadAd. AdView: " + bannerView + ", AdInfo: " + adInfo);
    }
    @Override
    public void adViewDidDisplayAd(@NonNull BannerView adView, AdInfo adInfo) {
        System.out.println("App - didDisplayAd. AdView: " + adView + ", AdInfo: " + adInfo);
    }

    @Override
    public void adViewDidFailToDisplayAd(@NonNull BannerView adView, AdInfo adInfo, Error errors) {
        System.out.println("App - didFailToDisplayAd. AdView: " + adView + ", Error: " + errors.getLocalizedMessage());
    }

    @Override
    public void adViewClicked(@NonNull BannerView adView, AdInfo adInfo) {
        System.out.println("App - didClicked. AdView: " + adView + ", AdInfo: " + adInfo);
    }

    @Override
    public void allNetworksFailedToDisplayAd(@NonNull BannerView adView) {
        System.out.println("App - didClicked. AdView: " + adView);
    }


}

File: bidappdemoapp-kotlin/app/src/main/java/com/bidapp/demo/BannerViewDelegate.kt
package com.bidapp.demo

import io.bidapp.sdk.AdInfo
import io.bidapp.sdk.BIDBannerViewDelegate
import io.bidapp.sdk.BannerView


class BannerViewDelegate() : BIDBannerViewDelegate {



    override fun adViewDidDisplayAd(adView: BannerView, adInfo: AdInfo?) {
        print("App - didDisplayAd. AdView: $adView, AdInfo: $adInfo")
    }

    override fun adViewDidFailToDisplayAd(adView: BannerView, adInfo: AdInfo?, errors: Error) {
        print("App - didFailToDisplayAd. AdView: $adView, Error:${errors.localizedMessage}")
    }

    override fun adViewDidLoadAd(adView: BannerView, adInfo: AdInfo?) {
        print("App - adViewDidLoadAd. AdView: $adView, AdInfo: $adInfo")
    }

    override fun allNetworksFailedToDisplayAd(adView: BannerView) {
        print("App - didFailToDisplayAd. AdView: $adView")
    }

    override fun adViewClicked(adView: BannerView, adInfo: AdInfo?) {
        print("App - didClicked. AdView: $adView, AdInfo: $adInfo")
    }



}